core/atomic_c11.c

```diff
@@ -43,8 +43,19 @@
 
 /* GCC documentation refers to the types as I1, I2, I4, I8, I16 */
 typedef uint8_t  I1;
+
+#if __SIZEOF_INT__==2
+typedef unsigned int I2;
+#else
 typedef uint16_t I2;
+#endif
+
+#if __SIZEOF_INT__==4
+typedef unsigned int I4;
+#else
 typedef uint32_t I4;
+#endif
```

Comment:
On Arch Linux using GCC 9.1.0 for AVR (docker image `archlinux/base`) this (using `unsigned int` instead of the matching `uintXX_t`)  seems not to be required. Can you re-check if this is really required?

To me it would make little sense if the compiler refuses to accept `uint16_t` instead of `unsigned int` on which `uint16_t` is `typedef`ed via `typedef unsigned int uint16_t` judging both common sense and my interpretation of the C standard (Section "6.7.7 Type definitions" in the C99 standard if you care to check).

Reply:
Correct. I've removed the typedef special handling for uint16_t.

So do I understand correctly that on ARM because uint32_t is typedef'ed as "long unsigned int" and that is considered a distinct type by the compiler?

Reply:
You're right. Running `$ echo __UINT32_TYPE__ | arm-none-eabi-gcc -E -` on Arch gives me:

```
# 1 "<stdin>"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "<stdin>"
long unsigned int
```

And `__UINT32_TYPE__` is used by newlib to define `uint32_t`, if defined. Otherwise it would have been defined as `unsigned int` by newlib and no special handling would be required.

Reply:
ok. I'll squash then. Need an ACK. :wink: